### PR TITLE
spec fixes

### PR DESCRIPTION
**Pull request by Joeysantoro**
Fixes withdraw/redeem to conform to most updated spec
Adds mint

fixes bug in calculateShares

### CODE REVIEW

File: src/mixins/ERC4626.sol [hunk 1]
```diff
@@ -59,48 +59,44 @@ abstract contract ERC4626 is ERC20 {
         afterDeposit(underlyingAmount);
     }
 
-    function withdraw(address to, uint256 underlyingAmount) public virtual returns (uint256 shares) {
-        _burn(msg.sender, shares = calculateShares(underlyingAmount));
+    function mint(address to, uint256 shareAmount) public virtual returns (uint256 underlyingAmount) {
+        _mint(to, shareAmount);
 
-        emit Withdraw(msg.sender, to, underlyingAmount);
+        emit Deposit(msg.sender, to, underlyingAmount = calculateUnderlying(shareAmount));
```
Comment (zerosnacks) [inline src/mixins/ERC4626.sol:65]: This is more of a code style preference but I would prefer it if `underlyingAmount = calculateUnderlying(shareAmount)` was defined outside of the `emit`, it is more explicit in my opinion seeing as we are using the underlyingAmount in multiple places after the event as well.

File: src/mixins/ERC4626.sol [hunk 2]
```diff
@@ -59,48 +59,44 @@ abstract contract ERC4626 is ERC20 {
         afterDeposit(underlyingAmount);
     }
 
-    function withdraw(address to, uint256 underlyingAmount) public virtual returns (uint256 shares) {
-        _burn(msg.sender, shares = calculateShares(underlyingAmount));
+    function mint(address to, uint256 shareAmount) public virtual returns (uint256 underlyingAmount) {
+        _mint(to, shareAmount);
```
Comment (zerosnacks) [inline src/mixins/ERC4626.sol:63]: I think this is incorrect, this should be `_mint(to, underlyingAmount = calculateUnderlying(shareAmount)`.
In the current implementation it always returns an `underlyingAmount` of 0.

File: src/mixins/ERC4626.sol (hunk 1 above)
Comment (zerosnacks) [reply src/mixins/ERC4626.sol:65]: (see comment below, this is actually part of a bug I think)

**Comment by zerosnacks** [comment]
LGTM 👍

**Comment by Joeysantoro** [comment]
@transmissions11 closing because this landed in solmate-next repo. We should double check and delete this branch to prevent confusion

**Pull request closed without merging**